velox/expression/tests/ExprCompilerTest.cpp

```diff
@@ -85,8 +86,27 @@ class ExprCompilerTest : public testing::Test,
   std::shared_ptr<core::QueryCtx> queryCtx_{std::make_shared<core::QueryCtx>()};
   std::unique_ptr<core::ExecCtx> execCtx_{
       std::make_unique<core::ExecCtx>(pool_.get(), queryCtx_.get())};
+
+  // Create constant expression from a variant with custom type.
+  std::shared_ptr<core::ConstantTypedExpr> makeConstantExprWithType(
+      std::shared_ptr<const Type> type,
+      variant value) {
+    return std::make_shared<core::ConstantTypedExpr>(
+        std::move(type), std::move(value));
+  }
 };
 
+TEST_F(ExprCompilerTest, constExprDecimalTypeCheck) {
```

Comment:
this test seems to succeed even without this change. Looks like the operator==  for ConstantTypedExpr checks for weak equivalence of types which also compares the precision and scale of the decimal and returns false. Can you please update the test with a representative test case?